src/GitHub.TeamFoundation.14/Connect/GitHubConnectSection.cs

```diff
@@ -344,20 +348,11 @@ public void Login()
         public bool OpenRepository()
         {
             var old = Repositories.FirstOrDefault(x => x.Equals(Holder.ActiveRepo));
-            // open the solution selection dialog when the user wants to switch to a different repo
-            // since there's no other way of changing the source control context in VS
             if (!Equals(SelectedRepository, old))
             {
-                if (ErrorHandler.Succeeded(ServiceProvider.GetSolution().OpenSolutionViaDlg(SelectedRepository.LocalPath, 1)))
-                {
-                    ServiceProvider.TryGetService<ITeamExplorer>()?.NavigateToPage(new Guid(TeamExplorerPageIds.Home), null);
-                    return true;
-                }
-                else
-                {
-                    SelectedRepository = old;
-                    return false;
-                }
+                vsServices.TryOpenRepository(SelectedRepository.LocalPath);
```

Comment:
In both cases we're using the side effect of opening a solution to change the repo context. If opening an empty solution fails, I can't see opening an existing solution working either.